In [1]:
# Day 2 - Data Cleaning

import pandas as pd


In [2]:
nav = pd.read_csv("../data/raw/02_nav_history.csv")

print("Shape:", nav.shape)
print("\nColumns:")
print(nav.columns)

print("\nMissing Values:")
print(nav.isnull().sum())

print("\nDuplicate Rows:")
print(nav.duplicated().sum())

nav.head()

Shape: (46000, 3)

Columns:
Index(['amfi_code', 'date', 'nav'], dtype='str')

Missing Values:
amfi_code    0
date         0
nav          0
dtype: int64

Duplicate Rows:
0


,amfi_code,date,nav
0,119551,2022-01-03,54.3856
1,119551,2022-01-04,54.3474
2,119551,2022-01-05,54.6869
3,119551,2022-01-06,55.4550
4,119551,2022-01-07,55.3692


In [4]:
print(nav.dtypes)

amfi_code      int64
date             str
nav          float64
dtype: object


In [5]:
nav['date'] = pd.to_datetime(nav['date'])

print(nav.dtypes)

amfi_code             int64
date         datetime64[us]
nav                 float64
dtype: object


In [6]:
#checking NAV values
print("Minimum NAV:", nav['nav'].min())
print("Maximum NAV:", nav['nav'].max())

Minimum NAV: 26.1366
Maximum NAV: 4268.5497


In [7]:
# Sort dataset
nav = nav.sort_values(
    ['amfi_code', 'date']
)

# Save cleaned file
nav.to_csv(
    "../data/processed/clean_nav_history.csv",
    index=False
)

print("clean_nav_history.csv saved successfully")

clean_nav_history.csv saved successfully


In [8]:
import os

print(
    os.path.exists(
        "../data/processed/clean_nav_history.csv"
    )
)

True


## Cleaning Summary - 02_nav_history.csv

- Rows: 46,000
- Missing Values: 0
- Duplicate Rows: 0
- Date converted to datetime
- NAV validated (>0)
- Dataset sorted by amfi_code and date
- Saved as clean_nav_history.csv

In [10]:
transactions = pd.read_csv(
    "../data/raw/08_investor_transactions.csv"
)

print("Shape:", transactions.shape)

print("\nColumns:")
print(transactions.columns)

print("\nMissing Values:")
print(transactions.isnull().sum())

print("\nDuplicate Rows:")
print(transactions.duplicated().sum())

transactions.head()

Shape: (32778, 13)

Columns:
Index(['investor_id', 'transaction_date', 'amfi_code', 'transaction_type',
       'amount_inr', 'state', 'city', 'city_tier', 'age_group', 'gender',
       'annual_income_lakh', 'payment_mode', 'kyc_status'],
      dtype='str')

Missing Values:
investor_id           0
transaction_date      0
amfi_code             0
transaction_type      0
amount_inr            0
state                 0
city                  0
city_tier             0
age_group             0
gender                0
annual_income_lakh    0
payment_mode          0
kyc_status            0
dtype: int64

Duplicate Rows:
0


,investor_id,transaction_date,amfi_code,transaction_type,amount_inr,state,city,city_tier,age_group,gender,annual_income_lakh,payment_mode,kyc_status
0,INV003054,2024-01-01,119092,SIP,1834,Telangana,Hyderabad,T30,56+,Female,77.1,UPI,Verified
1,INV002952,2024-01-01,148567,Redemption,392882,Punjab,Amritsar,B30,18-25,Male,7.1,Cheque,Verified
2,INV003420,2024-01-01,118636,SIP,912,Haryana,Faridabad,B30,36-45,Male,47.2,Mandate,Verified
3,INV003436,2024-01-01,118634,SIP,1102,Maharashtra,Mumbai,T30,36-45,Female,54.4,Cheque,Pending
4,INV004691,2024-01-01,119094,Lumpsum,8682,Delhi,Noida,T30,26-35,Male,14.5,Net Banking,Pending


In [11]:
print(transactions['transaction_type'].unique())

<ArrowStringArray>
['SIP', 'Redemption', 'Lumpsum']
Length: 3, dtype: str


In [12]:
print(transactions['kyc_status'].unique())

<ArrowStringArray>
['Verified', 'Pending']
Length: 2, dtype: str


In [13]:
print("Minimum Amount:", transactions['amount_inr'].min())
print("Maximum Amount:", transactions['amount_inr'].max())

Minimum Amount: 400
Maximum Amount: 597498


In [14]:
print(transactions.dtypes)

investor_id               str
transaction_date          str
amfi_code               int64
transaction_type          str
amount_inr              int64
state                     str
city                      str
city_tier                 str
age_group                 str
gender                    str
annual_income_lakh    float64
payment_mode              str
kyc_status                str
dtype: object


In [15]:
transactions['transaction_date'] = pd.to_datetime(
    transactions['transaction_date']
)

In [16]:
transactions.to_csv(
    "../data/processed/clean_investor_transactions.csv",
    index=False
)

print("clean_investor_transactions.csv saved")

clean_investor_transactions.csv saved


In [17]:
print("Transaction Types:")
print(transactions['transaction_type'].unique())

print("\nKYC Status:")
print(transactions['kyc_status'].unique())

print("\nMinimum Amount:")
print(transactions['amount_inr'].min())

Transaction Types:
<ArrowStringArray>
['SIP', 'Redemption', 'Lumpsum']
Length: 3, dtype: str

KYC Status:
<ArrowStringArray>
['Verified', 'Pending']
Length: 2, dtype: str

Minimum Amount:
400


## Cleaning Summary - 08_investor_transactions.csv

- Rows: 32,778
- Missing Values: 0
- Duplicate Rows: 0
- Transaction Types: SIP, Redemption, Lumpsum
- KYC Status: Verified, Pending
- Minimum Amount: 400
- transaction_date converted to datetime
- Saved as clean_investor_transactions.csv